mlx/array.h

```diff
@@ -266,7 +267,7 @@ class array {
   }
   // Check if the array is a tracer array
   bool is_tracer() const {
-    return array_desc_->is_tracer;
+    return array_desc_->is_tracer && currently_tracing();
   }
```

Comment:
Maybe cleaner to move this to `array.cpp` and include `transforms.h` there to avoid the forward declaration.